app/src/main/java/com/sdpteam/connectout/profile/ProfileFragment.java

```diff
@@ -59,19 +68,32 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         setupToolBar(ratingEditButton, toolbar, buttonText, listener);
 
         pvm.fetchProfile(userIdToDisplay);
-        pvm.getProfileLiveData().observe(getViewLifecycleOwner(), profile -> setTextViewsTo(view, profile));
+        pvm.getProfileLiveData().observe(getViewLifecycleOwner(), profile -> displayProfile(view, profile));
+
+        insertEventsListFragment(userIdToDisplay);
     }
 
-    private void setTextViewsTo(View view, Profile user) {
+    private void insertEventsListFragment(String userId) {
+        EventsViewModel viewModel = new ViewModelProvider(requireActivity(), new EventsViewModelFactory(new EventFirebaseDataSource())).get(EventsViewModel.class);
+        viewModel.setFilter(new EventParticipantIdFilter(userId), NONE);
+        EventsListViewFragment eventsListViewFragment = new EventsListViewFragment(viewModel);
+        getChildFragmentManager().beginTransaction().replace(R.id.profile_events_container, eventsListViewFragment).commit();
```

Comment:
same thing here! Wouldn't it be better to reuse the instances instead of creating a new one every time?